packages/builder/src/components/common/AppLockModal.svelte

```diff
@@ -0,0 +1,146 @@
+<script>
+  import {
+    Button,
+    ButtonGroup,
+    ModalContent,
+    Modal,
+    notifications,
+    ProgressCircle,
+  } from "@budibase/bbui"
+  import { auth, apps } from "stores/portal"
+  import { processStringSync } from "@budibase/string-templates"
+  import { API } from "api"
+
+  export let app
+  export let buttonSize = "M"
+
+  let APP_DEV_LOCK_SECONDS = 600 //common area for this?
+  let appLockModal
+  let processing = false
+
+  $: lockedBy = app?.lockedBy
+  $: lockedByYou = $auth.user.email === lockedBy?.email
+
+  $: lockIdentifer = `${
+    lockedBy && lockedBy.firstName ? lockedBy?.firstName : lockedBy?.email
+  }`
+
+  $: lockedByHeading =
+    lockedBy && lockedByYou ? "Locked by you" : `Locked by ${lockIdentifer}`
+
+  $: lockExpiry = getExpiryDuration(app)
```

Comment:
If we cache this in a reactive statement then the value will get stale over time, if the user doesn't refresh the page. I think we should just call `getExpiryDuration(app)` directly in the render code to ensure the label generated is always correct.

Reply:
Yep, that's fair enough. I've replaced the value in the modal.